Upstream-Status:Inappropriate [debian patch]

From: Niko Tyni <ntyni@debian.org>
Subject: Fix sprintf not to ignore LC_NUMERIC with constants
Bug-Debian: http://bugs.debian.org/601549
Bug: http://rt.perl.org/rt3/Ticket/Display.html?id=78632
Origin: upstream, http://perl5.git.perl.org/perl.git/commit/b3fd61496ebc585b1115807e3195f17714662a09

Don't fold constants in sprintf() if locales are used
    
An upstream regression in 5.10.1 made sprintf() ignore LC_NUMERIC for
numeric constants.

---
 op.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/op.c b/op.c
index e94f158..3c6badb 100644
--- a/op.c
+++ b/op.c
@@ -2503,6 +2503,7 @@ S_fold_constants(pTHX_ register OP *o)
     case OP_SLE:
     case OP_SGE:
     case OP_SCMP:
+    case OP_SPRINTF:
 	/* XXX what about the numeric ops? */
 	if (PL_hints & HINT_LOCALE)
 	    goto nope;
-- 
tg: (a508b62..) fixes/lc-numeric-sprintf (depends on: upstream)
